From c9b506cce9a3ae7e07a718c8e53ef3bd03285c61 Mon Sep 17 00:00:00 2001
From: Salvo Giangreco <giangrecosalvo9@gmail.com>
Date: Fri, 14 Nov 2025 19:35:05 +0100
Subject: [PATCH] Disable SemService

UN1CA: this patch is not complete! Please check unica/patches/ese/customize.sh for more info.
---
 .../com/android/server/SemService$1.smali     |   92 -
 .../server/SemService$SPITimeoutTask.smali    |   67 -
 .../SemServiceAccessControl$AllowList.smali   |  226 --
 .../SemServiceAccessControl$PackageList.smali |  236 --
 .../server/SemServiceAccessControl.smali      | 3108 -----------------
 .../com/android/server/SemServiceTools.smali  | 1148 ------
 .../samsung/android/ProductPackagesRune.smali |    2 +-
 .../SemService/SemServiceManager.smali        |    8 +-
 8 files changed, 4 insertions(+), 4883 deletions(-)
 delete mode 100644 smali_classes6/com/android/server/SemService$1.smali
 delete mode 100644 smali_classes6/com/android/server/SemService$SPITimeoutTask.smali
 delete mode 100644 smali_classes6/com/android/server/SemServiceAccessControl$AllowList.smali
 delete mode 100644 smali_classes6/com/android/server/SemServiceAccessControl$PackageList.smali
 delete mode 100644 smali_classes6/com/android/server/SemServiceAccessControl.smali
 delete mode 100644 smali_classes6/com/android/server/SemServiceTools.smali

diff --git a/smali_classes6/com/android/server/SemService$1.smali b/smali_classes6/com/android/server/SemService$1.smali
deleted file mode 100644
index c46b84b4..00000000
--- a/smali_classes6/com/android/server/SemService$1.smali
+++ /dev/null
@@ -1,92 +0,0 @@
-.class Lcom/android/server/SemService$1;
-.super Landroid/net/ConnectivityManager$NetworkCallback;
-.source "SemService.java"
-
-
-# annotations
-.annotation system Ldalvik/annotation/EnclosingMethod;
-    value = Lcom/android/server/SemService;->check_Network(I)V
-.end annotation
-
-.annotation system Ldalvik/annotation/InnerClass;
-    accessFlags = 0x0
-    name = null
-.end annotation
-
-
-# instance fields
-.field final synthetic blacklist this$0:Lcom/android/server/SemService;
-
-
-# direct methods
-.method constructor blacklist <init>(Lcom/android/server/SemService;)V
-    .locals 0
-    .annotation system Ldalvik/annotation/MethodParameters;
-        accessFlags = {
-            0x8010
-        }
-        names = {
-            null
-        }
-    .end annotation
-
-    iput-object p1, p0, Lcom/android/server/SemService$1;->this$0:Lcom/android/server/SemService;
-
-    invoke-direct {p0}, Landroid/net/ConnectivityManager$NetworkCallback;-><init>()V
-
-    return-void
-.end method
-
-
-# virtual methods
-.method public whitelist onAvailable(Landroid/net/Network;)V
-    .locals 2
-
-    new-instance v0, Ljava/lang/StringBuilder;
-
-    const-string v1, "F-NC : "
-
-    invoke-direct {v0, v1}, Ljava/lang/StringBuilder;-><init>(Ljava/lang/String;)V
-
-    invoke-virtual {v0, p1}, Ljava/lang/StringBuilder;->append(Ljava/lang/Object;)Ljava/lang/StringBuilder;
-
-    invoke-virtual {v0}, Ljava/lang/StringBuilder;->toString()Ljava/lang/String;
-
-    move-result-object p1
-
-    const-string v0, "SEC_ESE_Service"
-
-    invoke-static {v0, p1}, Landroid/util/Log;->e(Ljava/lang/String;Ljava/lang/String;)I
-
-    new-instance p1, Landroid/content/Intent;
-
-    const-string v0, "com.sec.action.SKMS_NETWORK"
-
-    invoke-direct {p1, v0}, Landroid/content/Intent;-><init>(Ljava/lang/String;)V
-
-    const-string v0, "com.sec.action.SKMS_NETWORK_VALUE"
-
-    const/4 v1, 0x1
-
-    invoke-virtual {p1, v0, v1}, Landroid/content/Intent;->putExtra(Ljava/lang/String;I)Landroid/content/Intent;
-
-    const-string v0, "com.skms.android.agent"
-
-    invoke-virtual {p1, v0}, Landroid/content/Intent;->setPackage(Ljava/lang/String;)Landroid/content/Intent;
-
-    iget-object p0, p0, Lcom/android/server/SemService$1;->this$0:Lcom/android/server/SemService;
-
-    invoke-static {p0}, Lcom/android/server/SemService;->-$$Nest$fgetmContext(Lcom/android/server/SemService;)Landroid/content/Context;
-
-    move-result-object p0
-
-    invoke-static {}, Landroid/os/Process;->myUserHandle()Landroid/os/UserHandle;
-
-    move-result-object v0
-
-    const-string v1, "com.samsung.permission.ESE_SYSTEM_PROTECTION"
-
-    invoke-virtual {p0, p1, v0, v1}, Landroid/content/Context;->sendBroadcastAsUser(Landroid/content/Intent;Landroid/os/UserHandle;Ljava/lang/String;)V
-
-    return-void
-.end method
diff --git a/smali_classes6/com/android/server/SemService$SPITimeoutTask.smali b/smali_classes6/com/android/server/SemService$SPITimeoutTask.smali
deleted file mode 100644
index 23ae4e46..00000000
--- a/smali_classes6/com/android/server/SemService$SPITimeoutTask.smali
+++ /dev/null
@@ -1,67 +0,0 @@
-.class Lcom/android/server/SemService$SPITimeoutTask;
-.super Ljava/util/TimerTask;
-.source "SemService.java"
-
-
-# annotations
-.annotation system Ldalvik/annotation/EnclosingClass;
-    value = Lcom/android/server/SemService;
-.end annotation
-
-.annotation system Ldalvik/annotation/InnerClass;
-    accessFlags = 0x0
-    name = "SPITimeoutTask"
-.end annotation
-
-
-# instance fields
-.field final synthetic blacklist this$0:Lcom/android/server/SemService;
-
-
-# direct methods
-.method constructor blacklist <init>(Lcom/android/server/SemService;)V
-    .locals 0
-    .annotation system Ldalvik/annotation/MethodParameters;
-        accessFlags = {
-            0x8010
-        }
-        names = {
-            null
-        }
-    .end annotation
-
-    iput-object p1, p0, Lcom/android/server/SemService$SPITimeoutTask;->this$0:Lcom/android/server/SemService;
-
-    invoke-direct {p0}, Ljava/util/TimerTask;-><init>()V
-
-    return-void
-.end method
-
-
-# virtual methods
-.method public whitelist test-api run()V
-    .locals 2
-
-    const-string v0, "SEC_ESE_Service"
-
-    const-string v1, "Close SPI if theree\'s no APDU communication in 30 seconds"
-
-    invoke-static {v0, v1}, Landroid/util/Log;->e(Ljava/lang/String;Ljava/lang/String;)I
-
-    iget-object v0, p0, Lcom/android/server/SemService$SPITimeoutTask;->this$0:Lcom/android/server/SemService;
-
-    const/4 v1, 0x0
-
-    invoke-virtual {v0, v1}, Lcom/android/server/SemService;->synchronizedCloseSpi(I)I
-
-    move-result v0
-
-    if-nez v0, :cond_0
-
-    iget-object p0, p0, Lcom/android/server/SemService$SPITimeoutTask;->this$0:Lcom/android/server/SemService;
-
-    invoke-static {p0}, Lcom/android/server/SemService;->-$$Nest$mreleaseSpiUsage(Lcom/android/server/SemService;)V
-
-    :cond_0
-    return-void
-.end method
diff --git a/smali_classes6/com/android/server/SemServiceAccessControl$AllowList.smali b/smali_classes6/com/android/server/SemServiceAccessControl$AllowList.smali
deleted file mode 100644
index cd169bb1..00000000
--- a/smali_classes6/com/android/server/SemServiceAccessControl$AllowList.smali
+++ /dev/null
@@ -1,226 +0,0 @@
-.class final Lcom/android/server/SemServiceAccessControl$AllowList;
-.super Ljava/lang/Object;
-.source "SemServiceAccessControl.java"
-
-
-# annotations
-.annotation system Ldalvik/annotation/EnclosingClass;
-    value = Lcom/android/server/SemServiceAccessControl;
-.end annotation
-
-.annotation system Ldalvik/annotation/InnerClass;
-    accessFlags = 0x1a
-    name = "AllowList"
-.end annotation
-
-
-# static fields
-.field private static final blacklist UID_NEED_TO_SET:I = -0x3e9
-
-.field private static final blacklist UID_NONE:I = -0x3e8
-
-
-# instance fields
-.field blacklist allowMap:Ljava/util/HashMap;
-    .annotation system Ldalvik/annotation/Signature;
-        value = {
-            "Ljava/util/HashMap<",
-            "Ljava/lang/String;",
-            "Ljava/util/ArrayList<",
-            "Ljava/lang/Integer;",
-            ">;>;"
-        }
-    .end annotation
-.end field
-
-
-# direct methods
-.method static bridge synthetic blacklist -$$Nest$mremove(Lcom/android/server/SemServiceAccessControl$AllowList;Ljava/lang/String;)V
-    .locals 0
-
-    invoke-direct {p0, p1}, Lcom/android/server/SemServiceAccessControl$AllowList;->remove(Ljava/lang/String;)V
-
-    return-void
-.end method
-
-.method public constructor blacklist <init>()V
-    .locals 1
-
-    invoke-direct {p0}, Ljava/lang/Object;-><init>()V
-
-    new-instance v0, Ljava/util/HashMap;
-
-    invoke-direct {v0}, Ljava/util/HashMap;-><init>()V
-
-    iput-object v0, p0, Lcom/android/server/SemServiceAccessControl$AllowList;->allowMap:Ljava/util/HashMap;
-
-    return-void
-.end method
-
-.method private blacklist remove(Ljava/lang/String;)V
-    .locals 0
-
-    iget-object p0, p0, Lcom/android/server/SemServiceAccessControl$AllowList;->allowMap:Ljava/util/HashMap;
-
-    invoke-virtual {p0, p1}, Ljava/util/HashMap;->remove(Ljava/lang/Object;)Ljava/lang/Object;
-
-    return-void
-.end method
-
-
-# virtual methods
-.method public blacklist add(Ljava/lang/String;)V
-    .locals 1
-
-    const/16 v0, -0x3e8
-
-    invoke-virtual {p0, p1, v0}, Lcom/android/server/SemServiceAccessControl$AllowList;->add(Ljava/lang/String;I)V
-
-    return-void
-.end method
-
-.method public blacklist add(Ljava/lang/String;I)V
-    .locals 2
-
-    iget-object v0, p0, Lcom/android/server/SemServiceAccessControl$AllowList;->allowMap:Ljava/util/HashMap;
-
-    new-instance v1, Ljava/util/ArrayList;
-
-    invoke-direct {v1}, Ljava/util/ArrayList;-><init>()V
-
-    invoke-virtual {v0, p1, v1}, Ljava/util/HashMap;->getOrDefault(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
-
-    move-result-object v0
-
-    check-cast v0, Ljava/util/ArrayList;
-
-    invoke-static {p2}, Ljava/lang/Integer;->valueOf(I)Ljava/lang/Integer;
-
-    move-result-object p2
-
-    invoke-virtual {v0, p2}, Ljava/util/ArrayList;->add(Ljava/lang/Object;)Z
-
-    iget-object p0, p0, Lcom/android/server/SemServiceAccessControl$AllowList;->allowMap:Ljava/util/HashMap;
-
-    invoke-virtual {p0, p1, v0}, Ljava/util/HashMap;->put(Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
-
-    return-void
-.end method
-
-.method public blacklist add(Ljava/lang/String;Ljava/lang/String;)V
-    .locals 1
-
-    const-class v0, Landroid/os/Process;
-
-    :try_start_0
-    invoke-virtual {v0, p2}, Ljava/lang/Class;->getField(Ljava/lang/String;)Ljava/lang/reflect/Field;
-
-    move-result-object p2
-
-    const/4 v0, 0x0
-
-    invoke-virtual {p2, v0}, Ljava/lang/reflect/Field;->get(Ljava/lang/Object;)Ljava/lang/Object;
-
-    move-result-object p2
-
-    check-cast p2, Ljava/lang/Integer;
-
-    invoke-virtual {p2}, Ljava/lang/Integer;->intValue()I
-
-    move-result p2
-
-    invoke-virtual {p0, p1, p2}, Lcom/android/server/SemServiceAccessControl$AllowList;->add(Ljava/lang/String;I)V
-    :try_end_0
-    .catch Ljava/lang/IllegalArgumentException; {:try_start_0 .. :try_end_0} :catch_0
-    .catch Ljava/lang/IllegalAccessException; {:try_start_0 .. :try_end_0} :catch_0
-    .catch Ljava/lang/NoSuchFieldException; {:try_start_0 .. :try_end_0} :catch_0
-    .catch Ljava/lang/SecurityException; {:try_start_0 .. :try_end_0} :catch_0
-
-    return-void
-
-    :catch_0
-    const/16 p2, -0x3e9
-
-    invoke-virtual {p0, p1, p2}, Lcom/android/server/SemServiceAccessControl$AllowList;->add(Ljava/lang/String;I)V
-
-    return-void
-.end method
-
-.method public blacklist match(Ljava/lang/String;I)Z
-    .locals 2
-
-    iget-object p0, p0, Lcom/android/server/SemServiceAccessControl$AllowList;->allowMap:Ljava/util/HashMap;
-
-    invoke-virtual {p0, p1}, Ljava/util/HashMap;->get(Ljava/lang/Object;)Ljava/lang/Object;
-
-    move-result-object p0
-
-    check-cast p0, Ljava/util/ArrayList;
-
-    const/4 v0, 0x0
-
-    if-nez p0, :cond_0
-
-    return v0
-
-    :cond_0
-    const/16 v1, -0x3e9
-
-    invoke-static {v1}, Ljava/lang/Integer;->valueOf(I)Ljava/lang/Integer;
-
-    move-result-object v1
-
-    invoke-virtual {p0, v1}, Ljava/util/ArrayList;->contains(Ljava/lang/Object;)Z
-
-    move-result v1
-
-    if-eqz v1, :cond_1
-
-    new-instance p0, Ljava/lang/StringBuilder;
-
-    invoke-direct {p0}, Ljava/lang/StringBuilder;-><init>()V
-
-    invoke-virtual {p0, p1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
-
-    const-string p1, "\'s UID need to set in android.os.Process!"
-
-    invoke-virtual {p0, p1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
-
-    invoke-virtual {p0}, Ljava/lang/StringBuilder;->toString()Ljava/lang/String;
-
-    move-result-object p0
-
-    const-string p1, "SEC_ESE_ServiceAccessControl"
-
-    invoke-static {p1, p0}, Landroid/util/Log;->e(Ljava/lang/String;Ljava/lang/String;)I
-
-    return v0
-
-    :cond_1
-    const/16 p1, -0x3e8
-
-    invoke-static {p1}, Ljava/lang/Integer;->valueOf(I)Ljava/lang/Integer;
-
-    move-result-object p1
-
-    invoke-virtual {p0, p1}, Ljava/util/ArrayList;->contains(Ljava/lang/Object;)Z
-
-    move-result p1
-
-    if-eqz p1, :cond_2
-
-    const/4 p0, 0x1
-
-    return p0
-
-    :cond_2
-    invoke-static {p2}, Ljava/lang/Integer;->valueOf(I)Ljava/lang/Integer;
-
-    move-result-object p1
-
-    invoke-virtual {p0, p1}, Ljava/util/ArrayList;->contains(Ljava/lang/Object;)Z
-
-    move-result p0
-
-    return p0
-.end method
diff --git a/smali_classes6/com/android/server/SemServiceAccessControl$PackageList.smali b/smali_classes6/com/android/server/SemServiceAccessControl$PackageList.smali
deleted file mode 100644
index a39ce5a6..00000000
--- a/smali_classes6/com/android/server/SemServiceAccessControl$PackageList.smali
+++ /dev/null
@@ -1,236 +0,0 @@
-.class public final enum Lcom/android/server/SemServiceAccessControl$PackageList;
-.super Ljava/lang/Enum;
-.source "SemServiceAccessControl.java"
-
-
-# annotations
-.annotation system Ldalvik/annotation/EnclosingClass;
-    value = Lcom/android/server/SemServiceAccessControl;
-.end annotation
-
-.annotation system Ldalvik/annotation/InnerClass;
-    accessFlags = 0x4019
-    name = "PackageList"
-.end annotation
-
-.annotation system Ldalvik/annotation/Signature;
-    value = {
-        "Ljava/lang/Enum<",
-        "Lcom/android/server/SemServiceAccessControl$PackageList;",
-        ">;"
-    }
-.end annotation
-
-
-# static fields
-.field private static final synthetic blacklist $VALUES:[Lcom/android/server/SemServiceAccessControl$PackageList;
-
-.field public static final enum blacklist MCosPatchPkgList:Lcom/android/server/SemServiceAccessControl$PackageList;
-
-.field public static final enum blacklist MFactoryPkgList:Lcom/android/server/SemServiceAccessControl$PackageList;
-
-.field public static final enum blacklist MFactoryResetList:Lcom/android/server/SemServiceAccessControl$PackageList;
-
-.field public static final enum blacklist MGrdmPkgList:Lcom/android/server/SemServiceAccessControl$PackageList;
-
-.field public static final enum blacklist MHWParamPkgList:Lcom/android/server/SemServiceAccessControl$PackageList;
-
-.field public static final enum blacklist MJavaPkgList:Lcom/android/server/SemServiceAccessControl$PackageList;
-
-.field public static final enum blacklist MLccmPkgList:Lcom/android/server/SemServiceAccessControl$PackageList;
-
-.field public static final enum blacklist MSKMSCardPkgList:Lcom/android/server/SemServiceAccessControl$PackageList;
-
-.field public static final enum blacklist MScpKmPkgList:Lcom/android/server/SemServiceAccessControl$PackageList;
-
-
-# direct methods
-.method private static synthetic blacklist $values()[Lcom/android/server/SemServiceAccessControl$PackageList;
-    .locals 9
-
-    sget-object v0, Lcom/android/server/SemServiceAccessControl$PackageList;->MJavaPkgList:Lcom/android/server/SemServiceAccessControl$PackageList;
-
-    sget-object v1, Lcom/android/server/SemServiceAccessControl$PackageList;->MScpKmPkgList:Lcom/android/server/SemServiceAccessControl$PackageList;
-
-    sget-object v2, Lcom/android/server/SemServiceAccessControl$PackageList;->MGrdmPkgList:Lcom/android/server/SemServiceAccessControl$PackageList;
-
-    sget-object v3, Lcom/android/server/SemServiceAccessControl$PackageList;->MSKMSCardPkgList:Lcom/android/server/SemServiceAccessControl$PackageList;
-
-    sget-object v4, Lcom/android/server/SemServiceAccessControl$PackageList;->MLccmPkgList:Lcom/android/server/SemServiceAccessControl$PackageList;
-
-    sget-object v5, Lcom/android/server/SemServiceAccessControl$PackageList;->MFactoryPkgList:Lcom/android/server/SemServiceAccessControl$PackageList;
-
-    sget-object v6, Lcom/android/server/SemServiceAccessControl$PackageList;->MHWParamPkgList:Lcom/android/server/SemServiceAccessControl$PackageList;
-
-    sget-object v7, Lcom/android/server/SemServiceAccessControl$PackageList;->MCosPatchPkgList:Lcom/android/server/SemServiceAccessControl$PackageList;
-
-    sget-object v8, Lcom/android/server/SemServiceAccessControl$PackageList;->MFactoryResetList:Lcom/android/server/SemServiceAccessControl$PackageList;
-
-    filled-new-array/range {v0 .. v8}, [Lcom/android/server/SemServiceAccessControl$PackageList;
-
-    move-result-object v0
-
-    return-object v0
-.end method
-
-.method static constructor blacklist <clinit>()V
-    .locals 3
-
-    new-instance v0, Lcom/android/server/SemServiceAccessControl$PackageList;
-
-    const-string v1, "MJavaPkgList"
-
-    const/4 v2, 0x0
-
-    invoke-direct {v0, v1, v2}, Lcom/android/server/SemServiceAccessControl$PackageList;-><init>(Ljava/lang/String;I)V
-
-    sput-object v0, Lcom/android/server/SemServiceAccessControl$PackageList;->MJavaPkgList:Lcom/android/server/SemServiceAccessControl$PackageList;
-
-    new-instance v0, Lcom/android/server/SemServiceAccessControl$PackageList;
-
-    const-string v1, "MScpKmPkgList"
-
-    const/4 v2, 0x1
-
-    invoke-direct {v0, v1, v2}, Lcom/android/server/SemServiceAccessControl$PackageList;-><init>(Ljava/lang/String;I)V
-
-    sput-object v0, Lcom/android/server/SemServiceAccessControl$PackageList;->MScpKmPkgList:Lcom/android/server/SemServiceAccessControl$PackageList;
-
-    new-instance v0, Lcom/android/server/SemServiceAccessControl$PackageList;
-
-    const-string v1, "MGrdmPkgList"
-
-    const/4 v2, 0x2
-
-    invoke-direct {v0, v1, v2}, Lcom/android/server/SemServiceAccessControl$PackageList;-><init>(Ljava/lang/String;I)V
-
-    sput-object v0, Lcom/android/server/SemServiceAccessControl$PackageList;->MGrdmPkgList:Lcom/android/server/SemServiceAccessControl$PackageList;
-
-    new-instance v0, Lcom/android/server/SemServiceAccessControl$PackageList;
-
-    const-string v1, "MSKMSCardPkgList"
-
-    const/4 v2, 0x3
-
-    invoke-direct {v0, v1, v2}, Lcom/android/server/SemServiceAccessControl$PackageList;-><init>(Ljava/lang/String;I)V
-
-    sput-object v0, Lcom/android/server/SemServiceAccessControl$PackageList;->MSKMSCardPkgList:Lcom/android/server/SemServiceAccessControl$PackageList;
-
-    new-instance v0, Lcom/android/server/SemServiceAccessControl$PackageList;
-
-    const-string v1, "MLccmPkgList"
-
-    const/4 v2, 0x4
-
-    invoke-direct {v0, v1, v2}, Lcom/android/server/SemServiceAccessControl$PackageList;-><init>(Ljava/lang/String;I)V
-
-    sput-object v0, Lcom/android/server/SemServiceAccessControl$PackageList;->MLccmPkgList:Lcom/android/server/SemServiceAccessControl$PackageList;
-
-    new-instance v0, Lcom/android/server/SemServiceAccessControl$PackageList;
-
-    const-string v1, "MFactoryPkgList"
-
-    const/4 v2, 0x5
-
-    invoke-direct {v0, v1, v2}, Lcom/android/server/SemServiceAccessControl$PackageList;-><init>(Ljava/lang/String;I)V
-
-    sput-object v0, Lcom/android/server/SemServiceAccessControl$PackageList;->MFactoryPkgList:Lcom/android/server/SemServiceAccessControl$PackageList;
-
-    new-instance v0, Lcom/android/server/SemServiceAccessControl$PackageList;
-
-    const-string v1, "MHWParamPkgList"
-
-    const/4 v2, 0x6
-
-    invoke-direct {v0, v1, v2}, Lcom/android/server/SemServiceAccessControl$PackageList;-><init>(Ljava/lang/String;I)V
-
-    sput-object v0, Lcom/android/server/SemServiceAccessControl$PackageList;->MHWParamPkgList:Lcom/android/server/SemServiceAccessControl$PackageList;
-
-    new-instance v0, Lcom/android/server/SemServiceAccessControl$PackageList;
-
-    const-string v1, "MCosPatchPkgList"
-
-    const/4 v2, 0x7
-
-    invoke-direct {v0, v1, v2}, Lcom/android/server/SemServiceAccessControl$PackageList;-><init>(Ljava/lang/String;I)V
-
-    sput-object v0, Lcom/android/server/SemServiceAccessControl$PackageList;->MCosPatchPkgList:Lcom/android/server/SemServiceAccessControl$PackageList;
-
-    new-instance v0, Lcom/android/server/SemServiceAccessControl$PackageList;
-
-    const-string v1, "MFactoryResetList"
-
-    const/16 v2, 0x8
-
-    invoke-direct {v0, v1, v2}, Lcom/android/server/SemServiceAccessControl$PackageList;-><init>(Ljava/lang/String;I)V
-
-    sput-object v0, Lcom/android/server/SemServiceAccessControl$PackageList;->MFactoryResetList:Lcom/android/server/SemServiceAccessControl$PackageList;
-
-    invoke-static {}, Lcom/android/server/SemServiceAccessControl$PackageList;->$values()[Lcom/android/server/SemServiceAccessControl$PackageList;
-
-    move-result-object v0
-
-    sput-object v0, Lcom/android/server/SemServiceAccessControl$PackageList;->$VALUES:[Lcom/android/server/SemServiceAccessControl$PackageList;
-
-    return-void
-.end method
-
-.method private constructor blacklist <init>(Ljava/lang/String;I)V
-    .locals 0
-    .annotation system Ldalvik/annotation/MethodParameters;
-        accessFlags = {
-            0x1000,
-            0x1000
-        }
-        names = {
-            null,
-            null
-        }
-    .end annotation
-
-    .annotation system Ldalvik/annotation/Signature;
-        value = {
-            "()V"
-        }
-    .end annotation
-
-    invoke-direct {p0, p1, p2}, Ljava/lang/Enum;-><init>(Ljava/lang/String;I)V
-
-    return-void
-.end method
-
-.method public static blacklist valueOf(Ljava/lang/String;)Lcom/android/server/SemServiceAccessControl$PackageList;
-    .locals 1
-    .annotation system Ldalvik/annotation/MethodParameters;
-        accessFlags = {
-            0x8000
-        }
-        names = {
-            null
-        }
-    .end annotation
-
-    const-class v0, Lcom/android/server/SemServiceAccessControl$PackageList;
-
-    invoke-static {v0, p0}, Ljava/lang/Enum;->valueOf(Ljava/lang/Class;Ljava/lang/String;)Ljava/lang/Enum;
-
-    move-result-object p0
-
-    check-cast p0, Lcom/android/server/SemServiceAccessControl$PackageList;
-
-    return-object p0
-.end method
-
-.method public static blacklist values()[Lcom/android/server/SemServiceAccessControl$PackageList;
-    .locals 1
-
-    sget-object v0, Lcom/android/server/SemServiceAccessControl$PackageList;->$VALUES:[Lcom/android/server/SemServiceAccessControl$PackageList;
-
-    invoke-virtual {v0}, [Lcom/android/server/SemServiceAccessControl$PackageList;->clone()Ljava/lang/Object;
-
-    move-result-object v0
-
-    check-cast v0, [Lcom/android/server/SemServiceAccessControl$PackageList;
-
-    return-object v0
-.end method
diff --git a/smali_classes6/com/android/server/SemServiceTools.smali b/smali_classes6/com/android/server/SemServiceTools.smali
deleted file mode 100644
index be0725eb..00000000
--- a/smali_classes6/com/android/server/SemServiceTools.smali
+++ /dev/null
@@ -1,1148 +0,0 @@
-.class public Lcom/android/server/SemServiceTools;
-.super Ljava/lang/Object;
-.source "SemServiceTools.java"
-
-
-# static fields
-.field public static final blacklist HEX_CHARS:[C
-
-.field private static final blacklist TAG:Ljava/lang/String; = "SEC_ESE_ServiceTools"
-
-.field private static final blacklist x_cord:[B
-
-.field private static final blacklist y_cord:[B
-
-
-# direct methods
-.method static constructor blacklist <clinit>()V
-    .locals 2
-
-    const/16 v0, 0x10
-
-    new-array v0, v0, [C
-
-    fill-array-data v0, :array_0
-
-    sput-object v0, Lcom/android/server/SemServiceTools;->HEX_CHARS:[C
-
-    const/16 v0, 0x20
-
-    new-array v1, v0, [B
-
-    fill-array-data v1, :array_1
-
-    sput-object v1, Lcom/android/server/SemServiceTools;->x_cord:[B
-
-    new-array v0, v0, [B
-
-    fill-array-data v0, :array_2
-
-    sput-object v0, Lcom/android/server/SemServiceTools;->y_cord:[B
-
-    return-void
-
-    :array_0
-    .array-data 2
-        0x30s
-        0x31s
-        0x32s
-        0x33s
-        0x34s
-        0x35s
-        0x36s
-        0x37s
-        0x38s
-        0x39s
-        0x41s
-        0x42s
-        0x43s
-        0x44s
-        0x45s
-        0x46s
-    .end array-data
-
-    :array_1
-    .array-data 1
-        0x47t
-        0x1et
-        0x70t
-        0x7ft
-        -0x75t
-        -0x64t
-        0x12t
-        0x21t
-        -0x80t
-        -0x73t
-        0x52t
-        0x66t
-        -0x70t
-        -0x27t
-        0x46t
-        0x6at
-        0x5t
-        0x2bt
-        0x3ct
-        0x2t
-        -0x58t
-        0x3et
-        0x55t
-        0x39t
-        -0x20t
-        -0x70t
-        0x15t
-        0x2dt
-        -0x72t
-        -0x73t
-        -0x9t
-        0x6et
-    .end array-data
-
-    :array_2
-    .array-data 1
-        0x12t
-        0xft
-        -0x36t
-        -0x2bt
-        0x4t
-        0x7et
-        -0xft
-        -0x5ft
-        -0x2bt
-        0x6at
-        0x2bt
-        0x1ct
-        0x4et
-        -0x55t
-        -0x1et
-        -0x7ct
-        -0x10t
-        0x6ft
-        -0x28t
-        -0x2dt
-        -0x68t
-        0x1dt
-        0x19t
-        -0x51t
-        -0x34t
-        0x22t
-        0x75t
-        0x64t
-        -0x3dt
-        -0x53t
-        -0x6dt
-        0x38t
-    .end array-data
-.end method
-
-.method public constructor blacklist <init>()V
-    .locals 0
-
-    invoke-direct {p0}, Ljava/lang/Object;-><init>()V
-
-    return-void
-.end method
-
-.method public static blacklist byteToHex(B)Ljava/lang/String;
-    .locals 3
-
-    sget-object v0, Lcom/android/server/SemServiceTools;->HEX_CHARS:[C
-
-    shr-int/lit8 v1, p0, 0x4
-
-    and-int/lit8 v1, v1, 0xf
-
-    aget-char v1, v0, v1
-
-    and-int/lit8 p0, p0, 0xf
-
-    aget-char p0, v0, p0
-
-    const/4 v0, 0x2
-
-    new-array v0, v0, [C
-
-    const/4 v2, 0x0
-
-    aput-char v1, v0, v2
-
-    const/4 v1, 0x1
-
-    aput-char p0, v0, v1
-
-    new-instance p0, Ljava/lang/String;
-
-    invoke-direct {p0, v0}, Ljava/lang/String;-><init>([C)V
-
-    return-object p0
-.end method
-
-.method public static blacklist bytesToHex([B)Ljava/lang/String;
-    .locals 6
-
-    if-eqz p0, :cond_1
-
-    array-length v0, p0
-
-    mul-int/lit8 v0, v0, 0x2
-
-    new-array v0, v0, [C
-
-    const/4 v1, 0x0
-
-    :goto_0
-    array-length v2, p0
-
-    if-ge v1, v2, :cond_0
-
-    mul-int/lit8 v2, v1, 0x2
-
-    sget-object v3, Lcom/android/server/SemServiceTools;->HEX_CHARS:[C
-
-    aget-byte v4, p0, v1
-
-    and-int/lit16 v5, v4, 0xf0
-
-    ushr-int/lit8 v5, v5, 0x4
-
-    aget-char v5, v3, v5
-
-    aput-char v5, v0, v2
-
-    add-int/lit8 v2, v2, 0x1
-
-    and-int/lit8 v4, v4, 0xf
-
-    aget-char v3, v3, v4
-
-    aput-char v3, v0, v2
-
-    add-int/lit8 v1, v1, 0x1
-
-    goto :goto_0
-
-    :cond_0
-    new-instance p0, Ljava/lang/String;
-
-    invoke-direct {p0, v0}, Ljava/lang/String;-><init>([C)V
-
-    return-object p0
-
-    :cond_1
-    const/4 p0, 0x0
-
-    return-object p0
-.end method
-
-.method public static blacklist ccmVerify([B[B)Z
-    .locals 5
-
-    const-string v0, "SEC_ESE_ServiceTools"
-
-    const/4 v1, 0x0
-
-    :try_start_0
-    const-string/jumbo v2, "verify start"
-
-    invoke-static {v0, v2}, Landroid/util/Log;->i(Ljava/lang/String;Ljava/lang/String;)I
-
-    if-eqz p0, :cond_3
-
-    array-length v2, p0
-
-    const/4 v3, 0x1
-
-    if-ge v2, v3, :cond_0
-
-    goto :goto_1
-
-    :cond_0
-    if-eqz p1, :cond_2
-
-    array-length v2, p1
-
-    const/16 v3, 0x40
-
-    if-eq v2, v3, :cond_1
-
-    goto :goto_0
-
-    :cond_1
-    const-string v2, "EC"
-
-    invoke-static {v2}, Ljava/security/KeyPairGenerator;->getInstance(Ljava/lang/String;)Ljava/security/KeyPairGenerator;
-
-    move-result-object v2
-
-    new-instance v3, Ljava/security/spec/ECGenParameterSpec;
-
-    const-string/jumbo v4, "secp256r1"
-
-    invoke-direct {v3, v4}, Ljava/security/spec/ECGenParameterSpec;-><init>(Ljava/lang/String;)V
-
-    invoke-virtual {v2, v3}, Ljava/security/KeyPairGenerator;->initialize(Ljava/security/spec/AlgorithmParameterSpec;)V
-
-    invoke-virtual {v2}, Ljava/security/KeyPairGenerator;->generateKeyPair()Ljava/security/KeyPair;
-
-    move-result-object v2
-
-    invoke-virtual {v2}, Ljava/security/KeyPair;->getPublic()Ljava/security/PublicKey;
-
-    move-result-object v2
-
-    check-cast v2, Ljava/security/interfaces/ECPublicKey;
-
-    invoke-interface {v2}, Ljava/security/interfaces/ECPublicKey;->getParams()Ljava/security/spec/ECParameterSpec;
-
-    move-result-object v2
-
-    sget-object v3, Lcom/android/server/SemServiceTools;->x_cord:[B
-
-    sget-object v4, Lcom/android/server/SemServiceTools;->y_cord:[B
-
-    invoke-static {v2, v3, v4}, Lcom/android/server/SemServiceTools;->decodeECPublicKey(Ljava/security/spec/ECParameterSpec;[B[B)Ljava/security/interfaces/ECPublicKey;
-
-    move-result-object v2
-
-    const-string v3, "SHA256withECDSA"
-
-    invoke-static {v3}, Ljava/security/Signature;->getInstance(Ljava/lang/String;)Ljava/security/Signature;
-
-    move-result-object v3
-
-    invoke-static {p1}, Lcom/android/server/SemServiceTools;->getAsnSignature([B)[B
-
-    move-result-object p1
-
-    invoke-virtual {v3, v2}, Ljava/security/Signature;->initVerify(Ljava/security/PublicKey;)V
-
-    invoke-virtual {v3, p0}, Ljava/security/Signature;->update([B)V
-
-    invoke-virtual {v3, p1}, Ljava/security/Signature;->verify([B)Z
-
-    move-result v1
-
-    goto/16 :goto_2
-
-    :cond_2
-    :goto_0
-    const-string/jumbo p0, "signature is invalid"
-
-    invoke-static {v0, p0}, Landroid/util/Log;->e(Ljava/lang/String;Ljava/lang/String;)I
-
-    return v1
-
-    :cond_3
-    :goto_1
-    const-string/jumbo p0, "message is invalid"
-
-    invoke-static {v0, p0}, Landroid/util/Log;->e(Ljava/lang/String;Ljava/lang/String;)I
-    :try_end_0
-    .catch Ljava/security/NoSuchAlgorithmException; {:try_start_0 .. :try_end_0} :catch_5
-    .catch Ljava/security/spec/InvalidKeySpecException; {:try_start_0 .. :try_end_0} :catch_4
-    .catch Ljava/security/InvalidKeyException; {:try_start_0 .. :try_end_0} :catch_3
-    .catch Ljava/security/SignatureException; {:try_start_0 .. :try_end_0} :catch_2
-    .catch Ljava/lang/NoClassDefFoundError; {:try_start_0 .. :try_end_0} :catch_1
-    .catch Ljava/security/InvalidAlgorithmParameterException; {:try_start_0 .. :try_end_0} :catch_0
-
-    return v1
-
-    :catch_0
-    move-exception p0
-
-    new-instance p1, Ljava/lang/StringBuilder;
-
-    const-string v2, "IAPE "
-
-    invoke-direct {p1, v2}, Ljava/lang/StringBuilder;-><init>(Ljava/lang/String;)V
-
-    invoke-virtual {p1, p0}, Ljava/lang/StringBuilder;->append(Ljava/lang/Object;)Ljava/lang/StringBuilder;
-
-    invoke-virtual {p1}, Ljava/lang/StringBuilder;->toString()Ljava/lang/String;
-
-    move-result-object p0
-
-    invoke-static {v0, p0}, Landroid/util/Log;->e(Ljava/lang/String;Ljava/lang/String;)I
-
-    goto :goto_2
-
-    :catch_1
-    move-exception p0
-
-    new-instance p1, Ljava/lang/StringBuilder;
-
-    const-string v2, "NCDFE "
-
-    invoke-direct {p1, v2}, Ljava/lang/StringBuilder;-><init>(Ljava/lang/String;)V
-
-    invoke-virtual {p1, p0}, Ljava/lang/StringBuilder;->append(Ljava/lang/Object;)Ljava/lang/StringBuilder;
-
-    invoke-virtual {p1}, Ljava/lang/StringBuilder;->toString()Ljava/lang/String;
-
-    move-result-object p0
-
-    invoke-static {v0, p0}, Landroid/util/Log;->e(Ljava/lang/String;Ljava/lang/String;)I
-
-    goto :goto_2
-
-    :catch_2
-    move-exception p0
-
-    new-instance p1, Ljava/lang/StringBuilder;
-
-    const-string v2, "SE "
-
-    invoke-direct {p1, v2}, Ljava/lang/StringBuilder;-><init>(Ljava/lang/String;)V
-
-    invoke-virtual {p1, p0}, Ljava/lang/StringBuilder;->append(Ljava/lang/Object;)Ljava/lang/StringBuilder;
-
-    invoke-virtual {p1}, Ljava/lang/StringBuilder;->toString()Ljava/lang/String;
-
-    move-result-object p0
-
-    invoke-static {v0, p0}, Landroid/util/Log;->e(Ljava/lang/String;Ljava/lang/String;)I
-
-    goto :goto_2
-
-    :catch_3
-    move-exception p0
-
-    new-instance p1, Ljava/lang/StringBuilder;
-
-    const-string v2, "IKE "
-
-    invoke-direct {p1, v2}, Ljava/lang/StringBuilder;-><init>(Ljava/lang/String;)V
-
-    invoke-virtual {p1, p0}, Ljava/lang/StringBuilder;->append(Ljava/lang/Object;)Ljava/lang/StringBuilder;
-
-    invoke-virtual {p1}, Ljava/lang/StringBuilder;->toString()Ljava/lang/String;
-
-    move-result-object p0
-
-    invoke-static {v0, p0}, Landroid/util/Log;->e(Ljava/lang/String;Ljava/lang/String;)I
-
-    goto :goto_2
-
-    :catch_4
-    move-exception p0
-
-    new-instance p1, Ljava/lang/StringBuilder;
-
-    const-string v2, "IKSE "
-
-    invoke-direct {p1, v2}, Ljava/lang/StringBuilder;-><init>(Ljava/lang/String;)V
-
-    invoke-virtual {p1, p0}, Ljava/lang/StringBuilder;->append(Ljava/lang/Object;)Ljava/lang/StringBuilder;
-
-    invoke-virtual {p1}, Ljava/lang/StringBuilder;->toString()Ljava/lang/String;
-
-    move-result-object p0
-
-    invoke-static {v0, p0}, Landroid/util/Log;->e(Ljava/lang/String;Ljava/lang/String;)I
-
-    goto :goto_2
-
-    :catch_5
-    move-exception p0
-
-    new-instance p1, Ljava/lang/StringBuilder;
-
-    const-string v2, "NSAE "
-
-    invoke-direct {p1, v2}, Ljava/lang/StringBuilder;-><init>(Ljava/lang/String;)V
-
-    invoke-virtual {p1, p0}, Ljava/lang/StringBuilder;->append(Ljava/lang/Object;)Ljava/lang/StringBuilder;
-
-    invoke-virtual {p1}, Ljava/lang/StringBuilder;->toString()Ljava/lang/String;
-
-    move-result-object p0
-
-    invoke-static {v0, p0}, Landroid/util/Log;->e(Ljava/lang/String;Ljava/lang/String;)I
-
-    :goto_2
-    new-instance p0, Ljava/lang/StringBuilder;
-
-    const-string/jumbo p1, "verify end : "
-
-    invoke-direct {p0, p1}, Ljava/lang/StringBuilder;-><init>(Ljava/lang/String;)V
-
-    invoke-virtual {p0, v1}, Ljava/lang/StringBuilder;->append(Z)Ljava/lang/StringBuilder;
-
-    invoke-virtual {p0}, Ljava/lang/StringBuilder;->toString()Ljava/lang/String;
-
-    move-result-object p0
-
-    invoke-static {v0, p0}, Landroid/util/Log;->i(Ljava/lang/String;Ljava/lang/String;)I
-
-    return v1
-.end method
-
-.method public static blacklist checkLength([BI)I
-    .locals 3
-
-    aget-byte v0, p0, p1
-
-    and-int/lit16 v1, v0, 0xff
-
-    const/16 v2, 0x80
-
-    if-ge v1, v2, :cond_0
-
-    return v0
-
-    :cond_0
-    const/16 v1, -0x7f
-
-    if-ne v0, v1, :cond_1
-
-    add-int/lit8 p1, p1, 0x1
-
-    aget-byte p0, p0, p1
-
-    and-int/lit16 p0, p0, 0xff
-
-    return p0
-
-    :cond_1
-    const/16 v1, -0x7e
-
-    if-ne v0, v1, :cond_2
-
-    add-int/lit8 v0, p1, 0x1
-
-    aget-byte v0, p0, v0
-
-    and-int/lit16 v0, v0, 0xff
-
-    shl-int/lit8 v0, v0, 0x8
-
-    add-int/lit8 p1, p1, 0x2
-
-    aget-byte p0, p0, p1
-
-    :goto_0
-    and-int/lit16 p0, p0, 0xff
-
-    add-int/2addr v0, p0
-
-    return v0
-
-    :cond_2
-    const/16 v1, -0x7d
-
-    if-ne v0, v1, :cond_3
-
-    add-int/lit8 v0, p1, 0x1
-
-    aget-byte v0, p0, v0
-
-    and-int/lit16 v0, v0, 0xff
-
-    shl-int/lit8 v0, v0, 0x10
-
-    add-int/lit8 v1, p1, 0x2
-
-    aget-byte v1, p0, v1
-
-    and-int/lit16 v1, v1, 0xff
-
-    shl-int/lit8 v1, v1, 0x8
-
-    add-int/2addr v0, v1
-
-    add-int/lit8 p1, p1, 0x3
-
-    aget-byte p0, p0, p1
-
-    goto :goto_0
-
-    :cond_3
-    const-string p0, "SEC_ESE_ServiceTools"
-
-    const-string p1, "Script Size Check error : -1"
-
-    invoke-static {p0, p1}, Landroid/util/Log;->e(Ljava/lang/String;Ljava/lang/String;)I
-
-    const/4 p0, -0x1
-
-    return p0
-.end method
-
-.method private static blacklist concatenate(BB)[B
-    .locals 3
-
-    const/4 v0, 0x1
-
-    new-array v1, v0, [B
-
-    const/4 v2, 0x0
-
-    aput-byte p0, v1, v2
-
-    new-array p0, v0, [B
-
-    aput-byte p1, p0, v2
-
-    const/4 p1, 0x2
-
-    new-array p1, p1, [B
-
-    invoke-static {v1, v2, p1, v2, v0}, Ljava/lang/System;->arraycopy(Ljava/lang/Object;ILjava/lang/Object;II)V
-
-    invoke-static {p0, v2, p1, v0, v0}, Ljava/lang/System;->arraycopy(Ljava/lang/Object;ILjava/lang/Object;II)V
-
-    return-object p1
-.end method
-
-.method private static blacklist concatenate(B[B)[B
-    .locals 3
-
-    const/4 v0, 0x1
-
-    new-array v1, v0, [B
-
-    const/4 v2, 0x0
-
-    aput-byte p0, v1, v2
-
-    if-nez p1, :cond_0
-
-    new-array p1, v2, [B
-
-    :cond_0
-    array-length p0, p1
-
-    add-int/2addr p0, v0
-
-    new-array p0, p0, [B
-
-    invoke-static {v1, v2, p0, v2, v0}, Ljava/lang/System;->arraycopy(Ljava/lang/Object;ILjava/lang/Object;II)V
-
-    array-length v1, p1
-
-    if-lez v1, :cond_1
-
-    array-length v1, p1
-
-    invoke-static {p1, v2, p0, v0, v1}, Ljava/lang/System;->arraycopy(Ljava/lang/Object;ILjava/lang/Object;II)V
-
-    :cond_1
-    return-object p0
-.end method
-
-.method private static blacklist concatenate([BB)[B
-    .locals 4
-
-    const/4 v0, 0x0
-
-    if-nez p0, :cond_0
-
-    new-array p0, v0, [B
-
-    :cond_0
-    const/4 v1, 0x1
-
-    new-array v2, v1, [B
-
-    aput-byte p1, v2, v0
-
-    array-length p1, p0
-
-    add-int/2addr p1, v1
-
-    new-array p1, p1, [B
-
-    array-length v3, p0
-
-    if-lez v3, :cond_1
-
-    array-length v3, p0
-
-    invoke-static {p0, v0, p1, v0, v3}, Ljava/lang/System;->arraycopy(Ljava/lang/Object;ILjava/lang/Object;II)V
-
-    :cond_1
-    array-length p0, p0
-
-    invoke-static {v2, v0, p1, p0, v1}, Ljava/lang/System;->arraycopy(Ljava/lang/Object;ILjava/lang/Object;II)V
-
-    return-object p1
-.end method
-
-.method private static blacklist concatenate([B[B)[B
-    .locals 3
-
-    const/4 v0, 0x0
-
-    if-nez p0, :cond_0
-
-    new-array p0, v0, [B
-
-    :cond_0
-    if-nez p1, :cond_1
-
-    new-array p1, v0, [B
-
-    :cond_1
-    array-length v1, p0
-
-    array-length v2, p1
-
-    add-int/2addr v1, v2
-
-    new-array v1, v1, [B
-
-    array-length v2, p0
-
-    if-lez v2, :cond_2
-
-    array-length v2, p0
-
-    invoke-static {p0, v0, v1, v0, v2}, Ljava/lang/System;->arraycopy(Ljava/lang/Object;ILjava/lang/Object;II)V
-
-    :cond_2
-    array-length v2, p1
-
-    if-lez v2, :cond_3
-
-    array-length p0, p0
-
-    array-length v2, p1
-
-    invoke-static {p1, v0, v1, p0, v2}, Ljava/lang/System;->arraycopy(Ljava/lang/Object;ILjava/lang/Object;II)V
-
-    :cond_3
-    return-object v1
-.end method
-
-.method private static blacklist decodeECPublicKey(Ljava/security/spec/ECParameterSpec;[B[B)Ljava/security/interfaces/ECPublicKey;
-    .locals 2
-    .annotation system Ldalvik/annotation/Throws;
-        value = {
-            Ljava/security/NoSuchAlgorithmException;,
-            Ljava/security/spec/InvalidKeySpecException;
-        }
-    .end annotation
-
-    new-instance v0, Ljava/math/BigInteger;
-
-    const/4 v1, 0x1
-
-    invoke-direct {v0, v1, p1}, Ljava/math/BigInteger;-><init>(I[B)V
-
-    new-instance p1, Ljava/math/BigInteger;
-
-    invoke-direct {p1, v1, p2}, Ljava/math/BigInteger;-><init>(I[B)V
-
-    new-instance p2, Ljava/security/spec/ECPoint;
-
-    invoke-direct {p2, v0, p1}, Ljava/security/spec/ECPoint;-><init>(Ljava/math/BigInteger;Ljava/math/BigInteger;)V
-
-    new-instance p1, Ljava/security/spec/ECPublicKeySpec;
-
-    invoke-direct {p1, p2, p0}, Ljava/security/spec/ECPublicKeySpec;-><init>(Ljava/security/spec/ECPoint;Ljava/security/spec/ECParameterSpec;)V
-
-    const-string p0, "EC"
-
-    invoke-static {p0}, Ljava/security/KeyFactory;->getInstance(Ljava/lang/String;)Ljava/security/KeyFactory;
-
-    move-result-object p0
-
-    invoke-virtual {p0, p1}, Ljava/security/KeyFactory;->generatePublic(Ljava/security/spec/KeySpec;)Ljava/security/PublicKey;
-
-    move-result-object p0
-
-    check-cast p0, Ljava/security/interfaces/ECPublicKey;
-
-    return-object p0
-.end method
-
-.method private static blacklist getAsnSignature([B)[B
-    .locals 7
-
-    const-string v0, "SEC_ESE_ServiceTools"
-
-    const/16 v1, 0x20
-
-    new-array v2, v1, [B
-
-    new-array v3, v1, [B
-
-    const/4 v4, 0x0
-
-    :try_start_0
-    invoke-static {p0, v4, v2, v4, v1}, Ljava/lang/System;->arraycopy(Ljava/lang/Object;ILjava/lang/Object;II)V
-
-    invoke-static {p0, v1, v3, v4, v1}, Ljava/lang/System;->arraycopy(Ljava/lang/Object;ILjava/lang/Object;II)V
-
-    move v1, v4
-
-    :goto_0
-    aget-byte v5, v2, v1
-
-    if-nez v5, :cond_0
-
-    add-int/lit8 v1, v1, 0x1
-
-    goto :goto_0
-
-    :cond_0
-    rsub-int/lit8 v5, v1, 0x20
-
-    new-array v6, v5, [B
-
-    invoke-static {v2, v1, v6, v4, v5}, Ljava/lang/System;->arraycopy(Ljava/lang/Object;ILjava/lang/Object;II)V
-
-    move v1, v4
-
-    :goto_1
-    aget-byte v2, v3, v1
-
-    if-nez v2, :cond_1
-
-    add-int/lit8 v1, v1, 0x1
-
-    goto :goto_1
-
-    :cond_1
-    rsub-int/lit8 v2, v1, 0x20
-
-    new-array v5, v2, [B
-
-    invoke-static {v3, v1, v5, v4, v2}, Ljava/lang/System;->arraycopy(Ljava/lang/Object;ILjava/lang/Object;II)V
-
-    aget-byte v1, v6, v4
-
-    and-int/lit16 v1, v1, 0xff
-
-    const/16 v2, 0x7f
-
-    if-le v1, v2, :cond_2
-
-    invoke-static {v4, v6}, Lcom/android/server/SemServiceTools;->concatenate(B[B)[B
-
-    move-result-object v6
-
-    :cond_2
-    aget-byte v1, v5, v4
-
-    and-int/lit16 v1, v1, 0xff
-
-    if-le v1, v2, :cond_3
-
-    invoke-static {v4, v5}, Lcom/android/server/SemServiceTools;->concatenate(B[B)[B
-
-    move-result-object v5
-
-    :cond_3
-    array-length v1, v6
-
-    int-to-byte v1, v1
-
-    const/4 v2, 0x2
-
-    invoke-static {v2, v1}, Lcom/android/server/SemServiceTools;->concatenate(BB)[B
-
-    move-result-object v1
-
-    invoke-static {v1, v6}, Lcom/android/server/SemServiceTools;->concatenate([B[B)[B
-
-    move-result-object v1
-
-    invoke-static {v1, v2}, Lcom/android/server/SemServiceTools;->concatenate([BB)[B
-
-    move-result-object v1
-
-    array-length v2, v5
-
-    int-to-byte v2, v2
-
-    invoke-static {v1, v2}, Lcom/android/server/SemServiceTools;->concatenate([BB)[B
-
-    move-result-object v1
-
-    invoke-static {v1, v5}, Lcom/android/server/SemServiceTools;->concatenate([B[B)[B
-
-    move-result-object v1
-
-    array-length v2, v1
-
-    int-to-byte v2, v2
-
-    invoke-static {v2, v1}, Lcom/android/server/SemServiceTools;->concatenate(B[B)[B
-
-    move-result-object v1
-
-    const/16 v2, 0x30
-
-    invoke-static {v2, v1}, Lcom/android/server/SemServiceTools;->concatenate(B[B)[B
-
-    move-result-object v1
-
-    new-instance v2, Ljava/lang/StringBuilder;
-
-    invoke-direct {v2}, Ljava/lang/StringBuilder;-><init>()V
-
-    const-string/jumbo v3, "raw: "
-
-    invoke-virtual {v2, v3}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
-
-    invoke-static {p0}, Lcom/android/server/SemServiceTools;->bytesToHex([B)Ljava/lang/String;
-
-    move-result-object p0
-
-    invoke-virtual {v2, p0}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
-
-    invoke-virtual {v2}, Ljava/lang/StringBuilder;->toString()Ljava/lang/String;
-
-    move-result-object p0
-
-    invoke-static {v0, p0}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
-
-    new-instance p0, Ljava/lang/StringBuilder;
-
-    invoke-direct {p0}, Ljava/lang/StringBuilder;-><init>()V
-
-    const-string v2, "encoded: "
-
-    invoke-virtual {p0, v2}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
-
-    invoke-static {v1}, Lcom/android/server/SemServiceTools;->bytesToHex([B)Ljava/lang/String;
-
-    move-result-object v2
-
-    invoke-virtual {p0, v2}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
-
-    invoke-virtual {p0}, Ljava/lang/StringBuilder;->toString()Ljava/lang/String;
-
-    move-result-object p0
-
-    invoke-static {v0, p0}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
-    :try_end_0
-    .catch Ljava/lang/Exception; {:try_start_0 .. :try_end_0} :catch_0
-
-    return-object v1
-
-    :catch_0
-    move-exception p0
-
-    invoke-virtual {p0}, Ljava/lang/Exception;->getMessage()Ljava/lang/String;
-
-    move-result-object p0
-
-    invoke-static {v0, p0}, Landroid/util/Log;->e(Ljava/lang/String;Ljava/lang/String;)I
-
-    const/4 p0, 0x0
-
-    return-object p0
-.end method
-
-.method public static blacklist getHexString([B)Ljava/lang/String;
-    .locals 4
-
-    new-instance v0, Ljava/lang/StringBuilder;
-
-    invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
-
-    const/4 v1, 0x0
-
-    :goto_0
-    array-length v2, p0
-
-    if-ge v1, v2, :cond_0
-
-    aget-byte v2, p0, v1
-
-    and-int/lit16 v2, v2, 0xff
-
-    add-int/lit16 v2, v2, 0x100
-
-    const/16 v3, 0x10
-
-    invoke-static {v2, v3}, Ljava/lang/Integer;->toString(II)Ljava/lang/String;
-
-    move-result-object v2
-
-    const/4 v3, 0x1
-
-    invoke-virtual {v2, v3}, Ljava/lang/String;->substring(I)Ljava/lang/String;
-
-    move-result-object v2
-
-    invoke-virtual {v0, v2}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
-
-    add-int/lit8 v1, v1, 0x1
-
-    goto :goto_0
-
-    :cond_0
-    invoke-virtual {v0}, Ljava/lang/StringBuilder;->toString()Ljava/lang/String;
-
-    move-result-object p0
-
-    return-object p0
-.end method
-
-.method public static blacklist getHexString([BII)Ljava/lang/String;
-    .locals 4
-
-    new-instance v0, Ljava/lang/StringBuilder;
-
-    invoke-direct {v0}, Ljava/lang/StringBuilder;-><init>()V
-
-    const/4 v1, 0x0
-
-    :goto_0
-    if-ge v1, p2, :cond_0
-
-    add-int v2, v1, p1
-
-    aget-byte v2, p0, v2
-
-    and-int/lit16 v2, v2, 0xff
-
-    add-int/lit16 v2, v2, 0x100
-
-    const/16 v3, 0x10
-
-    invoke-static {v2, v3}, Ljava/lang/Integer;->toString(II)Ljava/lang/String;
-
-    move-result-object v2
-
-    const/4 v3, 0x1
-
-    invoke-virtual {v2, v3}, Ljava/lang/String;->substring(I)Ljava/lang/String;
-
-    move-result-object v2
-
-    invoke-virtual {v0, v2}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
-
-    add-int/lit8 v1, v1, 0x1
-
-    goto :goto_0
-
-    :cond_0
-    invoke-virtual {v0}, Ljava/lang/StringBuilder;->toString()Ljava/lang/String;
-
-    move-result-object p0
-
-    return-object p0
-.end method
-
-.method public static blacklist hexToBytes(Ljava/lang/String;)[B
-    .locals 5
-
-    const/4 v0, 0x0
-
-    if-nez p0, :cond_0
-
-    return-object v0
-
-    :cond_0
-    invoke-virtual {p0}, Ljava/lang/String;->length()I
-
-    move-result v1
-
-    const/4 v2, 0x2
-
-    if-ge v1, v2, :cond_1
-
-    return-object v0
-
-    :cond_1
-    invoke-virtual {p0}, Ljava/lang/String;->length()I
-
-    move-result v0
-
-    div-int/2addr v0, v2
-
-    new-array v1, v0, [B
-
-    const/4 v2, 0x0
-
-    :goto_0
-    if-ge v2, v0, :cond_2
-
-    mul-int/lit8 v3, v2, 0x2
-
-    add-int/lit8 v4, v3, 0x2
-
-    invoke-virtual {p0, v3, v4}, Ljava/lang/String;->substring(II)Ljava/lang/String;
-
-    move-result-object v3
-
-    const/16 v4, 0x10
-
-    invoke-static {v3, v4}, Ljava/lang/Integer;->parseInt(Ljava/lang/String;I)I
-
-    move-result v3
-
-    int-to-byte v3, v3
-
-    aput-byte v3, v1, v2
-
-    add-int/lit8 v2, v2, 0x1
-
-    goto :goto_0
-
-    :cond_2
-    return-object v1
-.end method
-
-.method public static blacklist readFileBytes(Ljava/nio/file/Path;)Ljava/lang/String;
-    .locals 4
-
-    const-string v0, "SEC_ESE_ServiceTools"
-
-    const/4 v1, 0x0
-
-    :try_start_0
-    invoke-static {p0}, Ljava/nio/file/Files;->readAllBytes(Ljava/nio/file/Path;)[B
-
-    move-result-object p0
-    :try_end_0
-    .catch Ljava/io/IOException; {:try_start_0 .. :try_end_0} :catch_1
-    .catch Ljava/lang/Exception; {:try_start_0 .. :try_end_0} :catch_0
-
-    invoke-static {p0}, Lcom/android/server/SemServiceTools;->bytesToHex([B)Ljava/lang/String;
-
-    move-result-object p0
-
-    return-object p0
-
-    :catch_0
-    move-exception p0
-
-    new-instance v2, Ljava/lang/StringBuilder;
-
-    const-string v3, "Exception : "
-
-    invoke-direct {v2, v3}, Ljava/lang/StringBuilder;-><init>(Ljava/lang/String;)V
-
-    invoke-virtual {v2, p0}, Ljava/lang/StringBuilder;->append(Ljava/lang/Object;)Ljava/lang/StringBuilder;
-
-    invoke-virtual {v2}, Ljava/lang/StringBuilder;->toString()Ljava/lang/String;
-
-    move-result-object p0
-
-    invoke-static {v0, p0}, Landroid/util/Log;->e(Ljava/lang/String;Ljava/lang/String;)I
-
-    return-object v1
-
-    :catch_1
-    move-exception p0
-
-    new-instance v2, Ljava/lang/StringBuilder;
-
-    const-string v3, "IOException : "
-
-    invoke-direct {v2, v3}, Ljava/lang/StringBuilder;-><init>(Ljava/lang/String;)V
-
-    invoke-virtual {v2, p0}, Ljava/lang/StringBuilder;->append(Ljava/lang/Object;)Ljava/lang/StringBuilder;
-
-    invoke-virtual {v2}, Ljava/lang/StringBuilder;->toString()Ljava/lang/String;
-
-    move-result-object p0
-
-    invoke-static {v0, p0}, Landroid/util/Log;->e(Ljava/lang/String;Ljava/lang/String;)I
-
-    return-object v1
-.end method
diff --git a/smali_classes6/com/samsung/android/ProductPackagesRune.smali b/smali_classes6/com/samsung/android/ProductPackagesRune.smali
index 4dd32295..fdd45245 100644
--- a/smali_classes6/com/samsung/android/ProductPackagesRune.smali
+++ b/smali_classes6/com/samsung/android/ProductPackagesRune.smali
@@ -14,7 +14,7 @@
 
 .field public static final blacklist SECUREFOLDER:Z = true
 
-.field public static final blacklist SEM_DAEMON:Z = true
+.field public static final blacklist SEM_DAEMON:Z = false
 
 .field public static final blacklist SERVICE_SAMSUNG_BLOCKCHAIN:Z = true
 
diff --git a/smali_classes6/com/samsung/android/service/SemService/SemServiceManager.smali b/smali_classes6/com/samsung/android/service/SemService/SemServiceManager.smali
index f3e285e5..2653f5f4 100644
--- a/smali_classes6/com/samsung/android/service/SemService/SemServiceManager.smali
+++ b/smali_classes6/com/samsung/android/service/SemService/SemServiceManager.smali
@@ -68,7 +68,7 @@
 
 .field private static blacklist cosName:Ljava/lang/String; = null
 
-.field public static final blacklist isSupportSemService:Z = true
+.field public static final blacklist isSupportSemService:Z = false
 
 .field private static blacklist isSupportSemServiceManager:Z = false
 
@@ -125,16 +125,14 @@
 
     sput-object v0, Lcom/samsung/android/service/SemService/SemServiceManager;->HEX_CHARS:[C
 
-    const/4 v0, 0x1
+    const/4 v0, 0x0
 
     sput-boolean v0, Lcom/samsung/android/service/SemService/SemServiceManager;->isSupportSemServiceManager:Z
 
-    const-string v0, "JCOP6.3U"
+    const-string v0, ""
 
     sput-object v0, Lcom/samsung/android/service/SemService/SemServiceManager;->cosName:Ljava/lang/String;
 
-    const-string v0, "NXP"
-
     sput-object v0, Lcom/samsung/android/service/SemService/SemServiceManager;->chipVendor:Ljava/lang/String;
 
     return-void
-- 
2.51.2

